src/phases.ts

```diff
@@ -4483,7 +4483,7 @@ export class PokemonHealPhase extends CommonAnimPhase {
       this.scene.queueMessage(getPokemonMessage(pokemon, getStatusEffectHealText(lastStatusEffect)));
     }
 
-    if (fullHp && !lastStatusEffect) {
+    if (!(!fullHp || this.hpHealed < 0) && !lastStatusEffect) {
```

Comment:
Can this be simplified?
```suggestion
    if (fullHp && this.hpHealed >= 0 && !lastStatusEffect) {
```

Reply:
Intentionally made it the inverse of what is [here](https://github.com/pagefaultgames/pokerogue/blob/main/src/phases.ts#L4481) because the phase was either not ending at all or ending twice (which crashed the game) and I wanted to make it clear that those two are connected (it was also really easy to copy and paste it). I do accept it looks psychotic lmao and can probably just be reduced to a boolean. Need to double check tho

Reply:
made this a boolean so it doesn't look like crazy anymore